frontend/packages/console-app/src/components/volume-group-snapshot/create-volume-group-snapshot.tsx

```diff
@@ -0,0 +1,328 @@
+  ButtonBar,
+  getURLSearchParams,
+  HandlePromiseProps,
+  history,
+  ListDropdown,
+  PageHeading,
+  resourceObjPath,
+  withHandlePromise,
+} from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import {
+  PersistentVolumeClaimModel,
+  StorageClassModel,
+  VolumeGroupSnapshotClassModel,
+  VolumeGroupSnapshotModel,
+} from '@console/internal/models';
+import {
+  apiVersionForModel,
+  k8sCreate,
+  MatchExpression,
+  PersistentVolumeClaimKind,
+  referenceFor,
+  referenceForModel,
+  StorageClassResourceKind,
+  VolumeGroupSnapshotKind,
+} from '@console/internal/module/k8s';
+import { getNamespace } from '@console/shared';
+import './_create-volume-group-snapshot.scss';
+import { PVCTable } from './pvc-table';
+
+const VolumeGroupSnapshotClassDropdown: React.FC<SnapshotClassDropdownProps> = (props) => {
+  const { selectedKey } = props;
+  const kind = referenceForModel(VolumeGroupSnapshotClassModel);
+  const resources = [{ kind }];
+  const { t } = useTranslation();
+  return (
+    <ListDropdown
+      {...props}
+      desc={t('console-app~VolumeGroupSnapshotClass')}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder={t('console-app~Select volumegroup snapshot class')}
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const CreateGroupSnapshotForm = withHandlePromise<SnapshotResourceProps>((props) => {
+  const { namespace, handlePromise, inProgress, errorMessage } = props;
+  const { t } = useTranslation();
+  const [pvcObjs, setPVCObjs] = React.useState<PersistentVolumeClaimKind[]>([]);
+  const [groupSnapshotName, setGroupSnapshotName] = React.useState('group-snapshot');
+  const [groupSnapshotClassName, setGroupSnapshotClassName] = React.useState('');
+  const title = t('console-app~Create VolumeGroupSnapshot');
+  const [labelExpressions, setLabelExpressions] = React.useState<MatchExpression[]>([]);
+  const [labels, setLabels] = React.useState<{ [key: string]: string[] }>({});
+  const [validationError, setValidationError] = React.useState<string | null>(null);
+
+  const resourceWatch = React.useMemo(() => {
+    const watch = {
+      kind: referenceForModel(PersistentVolumeClaimModel),
+      isList: true,
+      namespace,
+      selector: labelExpressions.length > 0 ? { matchExpressions: labelExpressions } : undefined,
+    };
+
+    return watch;
+  }, [namespace, labelExpressions]);
+
+  const [data, loaded, loadError] = useK8sWatchResource<PersistentVolumeClaimKind[]>(resourceWatch);
+
+  const [storageClasses, setStorageClasses] = React.useState<StorageClassResourceKind[]>([]);
+  const storageClassWatch = {
+    kind: StorageClassModel.kind,
+    isList: true,
+  };
+
+  const [storageClassData] = useK8sWatchResource<StorageClassResourceKind[]>(storageClassWatch);
+
+  React.useEffect(() => {
+    if (storageClassData) {
+      setStorageClasses(storageClassData);
+    }
+  }, [storageClassData]);
```

Comment:
u don't need extra `useState` here, u can directly use `storageClassData`.
```suggestion
const [storageClassData] = useK8sWatchResource<StorageClassResourceKind[]>({
    kind: StorageClassModel.kind,
    isList: true,
  });
```